phc/easy/omics/genomic_test.py

```diff
@@ -110,14 +123,4 @@ def get_data_frame(
             **kw_args, **cls._get_current_args(inspect.currentframe(), locals())
         )
 
-        if test_type and len(df) > 0:
-            # TODO: Remove when API fixed
-
-            # NOTE: The API does not filter the returned sets because it is a
-            # nested structure. Since it's not a boatload of information, we opt
-            # to filter client-side for now.
-            return df[
-                df.setType == getattr(test_type, "value", test_type)
-            ].reset_index(drop=True)
```

Comment:
This transform code needs to be moved to the `transform_results` so that the caching doesn't save un-expanded JSON. If we do, we'll get problems since it won't be re-read as JSON (but instead as a string).